admin-api-server/src/nft/service/nft.service.ts

```diff
@@ -250,22 +251,31 @@ RETURNING id
 
   async deleteNft(user: UserEntity, nftId: number) {
     await this.nftLock.acquire(nftId);
-    const nft = await this.#findOne(nftId);
-    if (nft.createdBy !== user.id) {
-      throw new HttpException(
-        'no permission to delete this nft (only the creator may)',
-        HttpStatus.FORBIDDEN,
-      );
-    }
-    if (
-      !this.DELETABLE_IN_STATES.some((state: string) => nft.state === state)
-    ) {
-      throw new HttpException(
-        'no permission to delete this nft (nft is not in a state where it may still be deleted)',
-        HttpStatus.FORBIDDEN,
-      );
+
+    try {
+      const nft = await this.#findOne(nftId);
+      if (nft.createdBy !== user.id) {
+        throw new HttpException(
+          'no permission to delete this nft (only the creator may)',
+          HttpStatus.FORBIDDEN,
+        );
+      }
```

Comment:
I know we said only creator can delete the nft in the creation phase, but looking at this now make me wonder if adding the admin make sense as well

Reply:
Yeah I think it does too. I'll add it.